D1673/blas_interface.md

```diff
@@ -1691,160 +1808,337 @@ public:
   struct mapping {
   private:
     using nested_mapping_type =
-      typename Layout::template mapping<Extents>; // exposition only
+      typename Layout::template mapping<
+        transpose_extents_t<Extents>>; // exposition only
     nested_mapping_type nested_mapping_; // exposition only
+
   public:
-    mapping() = default;
+    mapping(const nested_mapping_type& map);
 
-    mapping(nested_mapping_type map) : nested_mapping_(map) {}
+    ptrdiff_t operator() (ptrdiff_t i, ptrdiff_t j) const
+      noexcept(noexcept(nested_mapping_(j, i)));
 
-    // ... insert other standard mapping things ...
+    nested_mapping_type nested_mapping() const;
 
-    ptrdiff_t operator() (ptrdiff_t i, ptrdiff_t j) const {
-      return nested_mapping_(j, i);
-    }
+    template<class OtherExtents>
+    bool operator==(const mapping<OtherExtents>& m) const;
 
-    nested_mapping() const { return nested_mapping_; }
+    template<class OtherExtents>
+    bool operator!=(const mapping<OtherExtents>& m) const;
+
+    Extents extents() const noexcept;
+
+    typename Extents::index_type required_span_size() const
+      noexcept(noexcept(nested_mapping_.required_span_size()));
+
+    bool is_unique() const
+      noexcept(noexcept(nested_mapping_.is_unique()));
+
+    bool is_contiguous() const
+      noexcept(noexcept(nested_mapping_.is_contiguous()));
+
+    bool is_strided() const
+      noexcept(noexcept(nested_mapping_.is_strided()));
+
+    static constexpr bool is_always_unique();
+
+    static constexpr bool is_always_contiguous();
+
+    static constexpr bool is_always_strided();
+
+    typename Extents::index_type
+    stride(typename Extents::index_type r) const
+      noexcept(noexcept(nested_mapping_.stride(r)));
   };
 };
 ```
 
+* *Requires:*
+
+  * `Layout` shall meet the `basic_mdspan` layout mapping policy
+    requirements (see *[mdspan.layout.reqs]* in P0009).
+
 * *Constraints:*
 
-  * `Layout` is a unique layout.
+  * `Layout::is_always_unique()` is `true`.
```

Comment:
Why do we need this? I would think we can just inherit this from the nested layout?

Reply:
@crtrott Ah, good point; thanks!

Reply:
@mhoemmen you want to change that first? Or shall I just merge this in and then we can change that later?

Reply:
@crtrott I'll just change it later -- I'll need to make a bunch of changes for `layout_blas_*` anyway.  Thanks!

Reply:
@crtrott Oh, wait, I get it.  `Layout` _is_ the nested layout.  This is a constraint on the nested layout.  I still need to fix this, though, because the nested layout does not define this -- the nested layout's mapping does.